lib/waterline/query/dql/tenant.js

```diff
@@ -0,0 +1,193 @@
+
+  if (arguments.length === 0) {
+    //
+    // Error
+    return usageError('Tenant must be passed.', usage, cb);
+  }
+
+  if (typeof tenant === 'function') {
+    // Tenant is required
+    // Error
+    return usageError('Tenant must be passed.', usage, cb);
+  }
+
+
+  // Change connection settings based on tenant
+  // console.log('collections', this.waterline.collections);
+  // console.log('adapterDictionary', this.waterline, this.adapterDictionary);
+
+  var baseCollectionIdentity = self.baseIdentity || self.identity;
+  var connectionName = this.adapterDictionary['identity'];
+  var baseConnectionName = connectionName;
+  if (baseCollectionIdentity != self.identity) {
+      // Is not the base Collection
+      // Get base collection
+      var baseCollection = self.waterline.collections[baseCollectionIdentity];
+    //   console.log('baseCollection', baseCollectionIdentity);
+      baseConnectionName = baseCollection.adapterDictionary['identity'];
+    //   console.log('baseConnectionName', baseConnectionName);
+  }
+  var connection = this.connections[connectionName];
+  // Change for Multi-Tenancy support
+  // console.log('connection', connection);
+  var config = connection.config;
+  // console.log('config', config);
+
+  if (config.isMultiTenant && typeof config.configForTenant === "function") {
+
+    var configForTenant = config.configForTenant;
+
+    // Return Deferred or pass to adapter
+    if (typeof cb === 'function') {
+
+      var configDup = _.cloneDeep(config);
+      configDup.tenant = tenant; // Set Tenant
+      // Check if it is the BaseConnection
+      if (connectionName === baseConnectionName) {
+          // Reset the tenant on the connection for the global/main Collection
+          config.tenant = null;
+      }
+      // Get the config for this tenant
+      configForTenant.call(self, tenant, configDup, function(err, config) {
+        // console.log('configForTenant', err, config);
+        if (err) return cb(err);
+
+        // Tenant's config namespace
+        var tenantSuffix = "<tenant:" + tenant + ">";
+        var newConnectionIdentity = (baseConnectionName + tenantSuffix).toLowerCase();
+
+        function createTenantCollection() {
+
+            // console.log('create tenant collection', self instanceof require('../../collection'), self);
+            var newCollectionIdentity = (baseCollectionIdentity + tenantSuffix).toLowerCase();
+
+            // Get TenantCollection, if already exists
+            var tenantCollection = self.waterline.collections[newCollectionIdentity];
+            if (tenantCollection) {
+                // There is already an existing TenantCollection
+                return cb(null, tenantCollection);
+            }
+            // No existing TenantCollection found
+            // Create a new TenantCollection!
+
+            var WaterlineCollection = require('../../collection');
+            var colDupModel = extend.call(WaterlineCollection, {
```

Comment:
Fixed with https://github.com/Glavin001/waterline/commit/afa2901e43cb1574f0bbe79cec9c1f9a51098c7c#diff-6ccdf090b069bfba17ecbc7d1ef471f2R89

That's everything I can think of.